Tools/Eflow/provision-eflow.ps1

```diff
@@ -1,109 +1,85 @@
+        Mikhail Chatillon       - chmikhai@microsoft.com
         Daniele Antonio Maggio  - daniele.maggio@microsoft.com
+        Nora Abi Akar           - noraabiakar@microsoft.com
 #>
 
 param
 (
-    [string]$switchName="EFLOW Switch",
-    [string]$startEflowIpRange=100,
-    [string]$endEflowIpRange=200,
     [string]$iotEdgeDeviceConnectionString,
+    [string]$switchName,
+    [string]$startEflowIpRange=100,
     [int]$internalPort=5000,
     [int]$externalPort=5000
 )
 
-# Enable Windows Features
-if ((Get-WindowsFeature -Name "DHCP").Installed -eq $false)
-{
-    Install-WindowsFeature -Name DHCP -IncludeManagementTools
-}
-if ((Get-WindowsFeature -Name "Hyper-V").Installed -eq $false)
+# Verify dependencies
+if (!$iotEdgeDeviceConnectionString)
 {
-    Install-WindowsFeature -Name Hyper-V
-    Install-WindowsFeature -Name Hyper-V-PowerShell
-    Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -All -NoRestart
-    Write-Host "Restarting the computer in 30 seconds to finish Hyper-V installation"
-    Start-Sleep -Seconds 30
-    Restart-Computer
+    throw "IoT Edge Device Connection String not provided."
 }
-if ((Get-WindowsFeature -Name "DHCP").Installed -eq $false)
+
+if (!$switchName)
 {
-    throw "DHCP not correctly installed."
+    throw "Switch Name not provided."
 }
+
 if ((Get-WindowsFeature -Name "Hyper-V").Installed -eq $false)
 {
     throw "Hyper-V not correctly installed."
 }
 
-# Create the networking
-if(Get-VMSwitch -Name $switchName -ErrorAction SilentlyContinue)
+if(!(Get-VMSwitch -Name $switchName -ErrorAction SilentlyContinue))
 {
-    throw "Switch already existing";
+    throw "Switch $switchName not found";
 }
 
-New-VMSwitch -Name $switchName -SwitchType internal
-Write-Host "Sleeping for 30sec before continuing for propagating the VMSwitch"
-Start-Sleep -Seconds 30
-Write-Host "Finished waiting for propagating the VMSwitch"
+# Create Nat
+Write-Host "Creating Nat"
+
 $netAdapterIfIndex=(Get-NetAdapter -Name "*$switchName*").ifIndex
 $netAdapterIpAddress=Get-NetIPAddress -AddressFamily IPv4  -InterfaceIndex $netAdapterIfIndex
 $netAdapterIp=$netAdapterIpAddress.IPAddress
 $ipAddressFamily=$netAdapterIp.Substring(0, $netAdapterIp.lastIndexOf('.')+1)
-$gwIpCounter=1
-$gwIp=$ipAddressFamily+$gwIpCounter
-
-while(!(get-NetIPAddress -IpAddress $gwIp -ErrorAction SilentlyContinue))
-{
-    $gwIpCounter++
-    if($gwIpCounter>9)
-    {
-        throw "All the IPs in the subnet range 1-9 on $ipAddressFamily were already taken";
-    }
-
-    $gwIp=$ipAddressFamily+$gwIpCounter
-}
-
+$gwIp=$ipAddressFamily+1
 $natIp=$ipAddressFamily+0
-
-New-NetIPAddress -IPAddress $gwIp -PrefixLength 24 -InterfaceIndex $netAdapterIfIndex
-
-if(Get-NetNat -Name "$switchName" -ErrorAction SilentlyContinue)
-{
-    throw "Net Nat with name $switchName already existing";
-}
+$startIp=$ipAddressFamily+$startEflowIpRange
 
 New-NetNat -Name "$switchName" -InternalIPInterfaceAddressPrefix "$natIp/24"
 
-#Install DHCP
```

Comment:
We don't need DHCP anymore? Are we relying on the DHCP from the network to which the VM is connected?

Reply:
I believe we never needed DHCP because we are assigning a static IP address to the EFLOW VM. @Mandur can you verify?

Reply:
By reading [this docs](https://learn.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows-networking?view=iotedge-1.4#ip-address-allocations) I was under the impression it was not needed as we need to use static IP to have a static nat mapping. Do you have another opinion @danigian ?

Reply:
All good, makes sense to not use DHCP for this scenario